vars/neoDeploy.groovy

```diff
@@ -111,123 +74,161 @@ void call(parameters = [:]) {
             stepParamKey3: 'scriptMissing',
             stepParam3: parameters?.script == null,
             stepParamKey4: 'legacyConfig',
-            stepParam4: ! stepCompatibilityConfiguration.isEmpty(),
+            stepParam4: !stepCompatibilityConfiguration.isEmpty(),
         ], configuration)
 
-        def archivePath = configuration.archivePath
-        if(archivePath?.trim()) {
-            if (!fileExists(archivePath)) {
-                error "Archive cannot be found with parameter archivePath: '${archivePath}'."
+        ToolDescriptor neo = new ToolDescriptor('SAP Cloud Platform Console Client', 'NEO_HOME', 'neoHome', '/tools/', 'neo.sh', null, 'version')
+        ToolDescriptor java = new ToolDescriptor('Java', 'JAVA_HOME', '', '/bin/', 'java', '1.8.0', '-version 2>&1')
+
+        if (configuration.neo.credentialsId) {
```

Comment:
If the credentials are not defined in Jenkins, the plugin will fail.

It is possible that you overwrite the credential parameter with null. Thus it would fail here.